frontend/src/components/Scratch/Scratch.tsx

```diff
@@ -182,13 +182,9 @@ export default function Scratch({
     }, [scratch.slug, scratch.last_updated])
 
     const cmExtensions = [...CODEMIRROR_EXTENSIONS, sourceCompareExtension]
+    if (useVim)
+        cmExtensions.push(vim())
     const cmExtensionsRef = useRef(cmExtensions)
```

Comment:
Why this ref? Feels like it should make the list of extensions never change from the first render, since the argument to useRef just serves as a default value

Reply:
So just use `cmExtensions` instead of creating `cmExtensionsRef` ?

Reply:
Yeah